proposals/maven_configuration_v2.md

```diff
@@ -0,0 +1,93 @@
+# Proposal: Align `jib-maven-plugin` configuration with `jib-gradle-plugin`
+
+## Motivation
+
+`jib-gradle-plugin` introduced a different configuration schema from `jib-maven-plugin`. This schema has many benefits over the original `jib-maven-plugin` configuration. Therefore, to maintain consistency between the two plugins, the `jib-maven-plugin` configuration should be updated to be more similar to the `jib-gradle-plugin` configuration.
+
+## Goals
+
+- Have `jib-maven-plugin` configuration be as similar to `jib-gradle-plugin` configuration as possible
+
+## Non-Goals
+
+- Maintain compatibility with older versions of `jib-maven-plugin` (we are still in alpha)
+
+## Current Configuration Schema
+
+The current `jib-maven-plugin` configuration looks like:
+
+```xml
+<configuration>
+  <from>openjdk:alpine</from>
+  <registry>localhost:5000</registry>
+  <repository>my-image</repository>
+  <tag>built-with-jib</tag>
+  <credHelpers>
+    <credHelper>osxkeychain</credHelper>
+  </credHelpers>
+  <jvmFlags>
+    <jvmFlag>-Xms512m</jvmFlag>
+    <jvmFlag>-Xdebug</jvmFlag>
+    <jvmFlag>-Xmy:flag=jib-rules</jvmFlag>
+  </jvmFlags>
+  <mainClass>mypackage.MyApp</mainClass>
+  <enableReproducibleBuilds>true</enableReproducibleBuilds>
+  <imageFormat>OCI</imageFormat>
+</configuration>
+```
+
+And the current `jib-gradle-plugin` configuration looks like:
+
+```groovy
+jib {
+  from {
+    image = 'openjdk:alpine'
+  }
+  to {
+    image = 'localhost:5000/my-image/built-with-jib'
+    credHelper = 'osxkeychain'
+  }
+  jvmFlags = ['-Xms512m', '-Xdebug', '-Xmy:flag=jib-rules']
+  mainClass = 'mypackage.MyApp'
+  reproducible = true
+  format = 'OCI'
+}
+```
+
+## Desired Configuration Schema
+
+Therefore, the desired `jib-maven-plugin` configuration should look like:
+
+```xml
+<configuration>
+  <from>
```

Comment:
I remember this was actually kind of a pain to do? Have you looked at the implementation of it?

Reply:
I think this can just be done with a class as the parameter type.

For example, in `docker-maven-plugin`: https://github.com/fabric8io/docker-maven-plugin/blob/master/src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java#L166\

The nested configuration class: https://github.com/fabric8io/docker-maven-plugin/blob/master/src/main/java/io/fabric8/maven/docker/config/RegistryAuthConfiguration.java

Reply:
oh weird, so `@Parameter` will implictly trigger nested setting of parameters? and we can set defaults on the inner members?

Reply:
Yeah, it might need to implement `Serializable` too.